### PR TITLE
Stage Nomad opt-in banner on .io site

### PR DESCRIPTION
**Pull request by ashleemboyer**
## 🔗 Relevant links

- Asana tasks 🎟️ 
  - [Stage adding "nomad" production beta_product_slugs array](https://app.asana.com/0/1202097197789424/1202632199481601/f)
  - [[DevDotOptIn] Cookie is not correctly set on VaultProject.io](https://app.asana.com/0/1202097197789424/1202699829872281/f)

## 🗒️ What

- Adds `ProductMetaProvider` to `NomadIoLayout`
- Adds `optInBanner` prop passed to `DocsPageInner` by `NomadDocsLandingPage`
- Delegates rendering logic of `DevDotOptIn` to the body of the component since the logic is always the same
- Updates `DevDotOptIn` to use a `<a>` instead of a `<button>` since clicking the element causes navigation
- Fixes the cookie set by `DevDotOptIn` to no longer be hardcoded to `"waypoint"` (this is a bug)

### Vault

- [ ] Go to [the preview URL](https://dev-portal-git-ambstage-nomad-io-banner-hashicorp.vercel.app/)
- [ ] Choose Vault from the product switcher at the bottom left
- [ ] For the /docs, /api-docs routes:
  - [ ] Go to the page
  - [ ] Make sure there is no "vault-io-beta-opt-in" cookie set
  - [ ] Make sure there is no "waypoint-io-beta-opt-in" cookie set
  - [ ] Make sure the location of the "Join the beta" link is correct
  - [ ] Click the "Join the beta" link
  - [ ] You should be taken to the corresponding developer.hashicorp.com page with the `?optInFrom=vault-io` search query
  - [ ] Go back to the .io preview page you were on
  - [ ] Make sure the "vault-io-beta-opt-in" cookie is set
  - [ ] Make sure there is no "waypoint-io-beta-opt-in" cookie set

### Waypoint

- [ ] Go to [the preview URL](https://dev-portal-git-ambstage-nomad-io-banner-hashicorp.vercel.app/)
- [ ] Choose Waypoint from the product switcher at the bottom left
- [ ] For the /docs, /commands, /plugins routes:
  - [ ] Go to the page
  - [ ] Make sure there is no "waypoint-io-beta-opt-in" cookie set
  - [ ] Make sure the location of the "Join the beta" link is correct
  - [ ] Click the "Join the beta" link
  - [ ] You should be taken to the corresponding developer.hashicorp.com page with the `?optInFrom=waypoint-io` search query
  - [ ] Go back to the .io preview page you were on
  - [ ] Make sure the "waypoint-io-beta-opt-in" cookie is set

### Nomad

- [ ] Go to [the preview URL](https://dev-portal-git-ambstage-nomad-io-banner-hashicorp.vercel.app/)
- [ ] Choose Nomad from the product switcher at the bottom left
- [ ] For the /docs, /api-docs, /plugins, /tools routes:
  - [ ] Go to the page
  - [ ] Make sure there is no "nomad-io-beta-opt-in" cookie set
  - [ ] Make sure the location of the "Join the beta" link is correct
  - [ ] Click the "Join the beta" link
  - [ ] You should be taken to the corresponding developer.hashicorp.com page with the `?optInFrom=nomad-io` search query
    - NOTE: the 404 error will show but that's OK since Nomad isn't in production beta yet 
  - [ ] Go back to the /docs page
  - [ ] Make sure the "nomad-io-beta-opt-in" cookie is set

## 💭 Anything else?

The approach I've taken here allows us to merge this PR to fix bugs without affecting the production NomadProject.io website. We'll want to create a separate PR just for adding `"nomad"` to the beta product slugs array.

### CODE REVIEW

File: src/components/_proxied-dot-io/common/dev-dot-opt-in/index.tsx [hunk 1]
```diff
@@ -22,30 +25,45 @@ const getDevDotLink = (product, path) => {
...
+	if (!shouldRenderOptInCTA) {
+		return null
+	}
+
+	function handleOptIn() {
 		// Set a cookie to ensure any future navigation will send them to dev dot
-		Cookies.set('waypoint-io-beta-opt-in', true, {
+		Cookies.set(`${productSlug}-io-beta-opt-in`, true, {
 			expires: DAYS_UNTIL_EXPIRE,
 		})
```
Comment (BRKalow) [inline src/components/_proxied-dot-io/common/dev-dot-opt-in/index.tsx:51]: 🤦‍♂️ good catch haha

**Review by BRKalow — COMMENTED** [review]
@ashleemboyer I'm not seeing the CTA on nomad's `/api-docs`, `/tools`, or `/plugins` routes. 💭 

We might need to add the `ProductMetaProvider` to nomad's layout file, as is done for vault here: https://github.com/hashicorp/dev-portal/blob/main/src/layouts/_proxied-dot-io/vault/index.tsx#L46

**Comment by ashleemboyer** [comment]
> @ashleemboyer I'm not seeing the CTA on nomad's `/api-docs`, `/tools`, or `/plugins` routes. 💭
> 
> We might need to add the `ProductMetaProvider` to nomad's layout file, as is done for vault here: https://github.com/hashicorp/dev-portal/blob/main/src/layouts/_proxied-dot-io/vault/index.tsx#L46

Oh! I didn't check those routes. I'll fix shortly 😊

**Comment by ashleemboyer** [comment]
@BRKalow should be fixed now! 👍🏻

**Review by BRKalow — APPROVED** [review]
LGTM!

**Pull request merged**